[WIP] Terraform Docs for TF policy (#2690)
hashicorp/web-unified-docs: content/terraform-docs-common/docs/cloud-docs/workspaces/policy-enforcement/define-policies/index.mdx

## Workflows

> You can use three policy-as-code frameworks to define fine-grained, logic-based policies: Terraform policy, Sentinel, or Open Policy Agent (OPA). Depending on the settings, policies can act as advisory warnings or firm requirements that prevent Terraform from provisioning infrastructure.

Reviewer: I think we have to put a beta disclaimer for tfpolicy: "Terraform policy (beta)"

Reviewer 2: ```suggestion
You can use the following policy-as-code frameworks to define fine-grained, logic-based policies. Depending on the settings, policies can act as advisory warnings or firm requirements that prevent Terraform from provisioning infrastructure:
```

> - **Terraform policy (beta):** You define policies using HCL (the same language as Terraform) to define policies. This makes it familiar to Terraform users and allows for native integration with Terraform workflows. Refer to [Define Terraform policy policies](/terraform/cloud-docs/policy-enforcement/define-policies/terraform-policy) for details.

Reviewer 2: ```suggestion
- **Terraform policy (beta):** Define policies in HCL. Because Terraform configuration language also uses HCL syntax, it should be familiar to infrastructure-as-code developers. The common syntax also allows you to natively integrate with Terraform workflows. Refer to [Define Terraform policy](/terraform/cloud-docs/policy-enforcement/define-policies/terraform-policy) for details.
```

- **Sentinel:** You define policies with the [Sentinel policy language](/sentinel/docs/concepts/language) and use imports to parse the Terraform plan, state, and configuration. Refer to [Define custom Sentinel policies](/terraform/cloud-docs/policy-enforcement/define-policies/custom-sentinel) for details.
- **OPA:** You define policies with the [Rego policy language](https://www.openpolicyagent.org/docs/latest/policy-language/). Refer to [Defining OPA Policies](/terraform/cloud-docs/policy-enforcement/define-policies/opa) for details.

> | Feature | Terraform policy | Sentinel | OPA |

Reviewer 2: ```suggestion
The following table compares select characteristics of the supported policy frameworks. For a more detailed comparison, refer to [Compare policy frameworks](/terraform/policy/compare).

| Characteristic | Terraform policy | Sentinel | OPA |
```
I don't know if "characteristic" is the right word, but I don't think "feature" is. We tend to use "feature" to mean "capability" or "functionality", but "learning curve" and "minimum Terraform version" aren't features in the same sense.

|---------|------------------|----------|-----|
| Language | HCL | Sentinel | Rego |

> | Learning curve | Low (if you know Terraform) | Medium | High |

Reviewer 2: ```suggestion
| Language complexity | Low for Terraform users | Medium | High |
```
I get what we are trying to communicate, but we want to avoid editorializing on the perceived level of difficulty. I think it would be better to remove this row because we say something very similar in line 18.

> | Minimum Terraform version | v1.0+ | - | - |

Reviewer: Minimum Terraform version that a workspace must be upgraded to in order to support a tfpolicy evaluation is [terraform_1.16.0-alpha20260706]

> | Minimum Terraform version | v1.16.0-alpha20260706+ | - | - |

Reviewer 3: Ignore the previous suggestion. The latest alpha version is now `
v1.16.0-alpha20260708` .

> | Minimum Terraform version | v1.16.0-alpha20260706+ | - | - |

Reviewer 2: ```suggestion
```
Not an appropriate comparison unless we wanted to include minimum versions of Sentinel and OPA we support, but I don't know that it would add value.

| Native Terraform integration | Yes | No | No |

> | Testing Framework | Built-in | Built-in | External |

Reviewer 2: ```suggestion
| Testing framework | Built-in | Built-in | External |
```
